Add browser template docs

meyerjrr opened the pull request:
In this PR I also updated the configuration docs to be separated out into different pages to help with discoverability

chrnorm commented on granted/configuration.mdx:
```diff
@@ -1,255 +0,0 @@
----
```
We need to add a redirect here to avoid breaking links, in case any users have linked to this page: https://mintlify.com/docs/settings/broken-links#redirects

chrnorm commented on granted/configuration/misc.mdx:
```diff
@@ -0,0 +1,25 @@
+---
+title: "Miscellaneous"
```
this page would need a clearer name, `misc` is confusing. Some of the content here may be better shifted to an "internals" page.

chrnorm reviewed (changes requested):
I think we should split this into two separate PRs: one to add the browser template guide and a second PR to improve our config docs.

I'm not sure if the approach in this PR of splitting the config page up into a subfolder is ideal. We currently have a "Usage" section and a "Recipes" section. A lot of the content in the Config section may be better placed inside of "Usage"? (and we should consider removing Recipes, perhaps?)

What I'd propose we do instead is reduce the content in the configuration page, and have it just focused on the `~/.granted/config` file. e.g. we should show an example config, and explain what each section in the config does. Everything else should be shifted to Usage.

meyerjrr commented:
Thanks @chrnorm good feedback. I have altered the change to move all the new pages to under the usage folder + removed the recipes section and moved those pages to under usage as well. 

I think this is much easier to find all the usage features in one place

Another thing I was thinking was making integrations with Common Fate easier to find. There has been a couple times where users were unable to find the relevant documentation and resorted to asking us directly. 
I have moved these under in a subfolder, and added documentation for creating profiles with the HTTP registry feature
![Screenshot 2024-09-25 at 5 33 11 PM](https://github.com/user-attachments/assets/8b7f724e-6fe7-4fa9-baf0-d5074cb02a5e)

Keen for your thoughts here. Happy to scrap it / move it into a separate PR if you'd like that as well.
